samples/msgext-northwind-inventory-python/lab/Exercise 02 - Run sample app.md

```diff
@@ -0,0 +1,260 @@
+# Building Message Extensions for Microsoft Copilot for Microsoft 365
+
+TABLE OF CONTENTS
+
+* [Welcome](./Exercise%2000%20-%20Welcome.md) 
+* [Exercise 1](./Exercise%2001%20-%20Set%20up.md) - Set up your development Environment 
+* Exercise 2 - Run the sample as a Message Extension **(THIS PAGE)**
+* [Exercise 3](./Exercise%2003%20-%20Run%20in%20Copilot.md) - Run the sample as a Copilot plugin
+* [Exercise 4](./Exercise%2004%20-%20Add%20a%20new%20command.md) - Add a new command
+* [Exercise 5](./Exercise%2005%20-%20Code%20tour.md) - Code tour
+
+## Exercise 2 - Run the sample as a Message Extension
+
+## Step 1 - Set up the project for first use
+
+Open your working folder in Visual Studio Code.
+
+Teams Toolkit stores environment variables in the **env** folder, and it will fill in all the values automatically when you start your project the first time. However there's one value that's specific to the sample application, and that's the connection string for accessing the Northwind database.
+
+In this project, the Northwind database is stored in Azure Table Storage; when you're debugging locally, it uses the [Azurite](https://learn.microsoft.com/azure/storage/common/storage-use-azurite?tabs=visual-studio) storage emulator.
+
+Install azurite using npm command:
+
+~~~
+npm install -g azurite
+~~~
+
+![npm install](./images/02-01-npm-install-azurite-01.png)
+
+Start azurite using below npm command:
+
+~~~
+azurite --silent --location c:\azurite --debug c:\azurite\debug.log
+~~~
+
+![npm start](./images/02-01-npm-start-azurite-01.png)
+
+The necessary setting is provided in a file **.env.sample**. Make a copy of this file, and call it **.env**. This is where secret or sensitive settings are stored.
+
+If you're not sure how to do this, here are the steps in Visual Studio Code. Copy the value `STORAGE_ACCOUNT_CONNECTION_STRING=UseDevelopmentStorage=true`in **.env.sample** and paste it to the **.env** file
+
+The resulting **.env** file should contain this line:
+
+~~~text
+STORAGE_ACCOUNT_CONNECTION_STRING=UseDevelopmentStorage=true
+~~~
+
+![Copy .env.sample to .env](./images/02-01-Setup-Project-01.png)
```

Comment:
These steps can be automated by including them in the F5 experience.

I suggest that we:

1. Remove the `.env.sample` file from root
1. Create a new file in `env` folder called `.env.local.user` with the following contents:

	```text
	SECRET_STORAGE_ACCOUNT_CONNECTION_STRING=UseDevelopmentStorage=true
	```
1. Update `file/createOrUpdateEnvironmentFile` action in `teamsapp.local.yml` to write the environment variable value to the `.env` file in the root.

	```yml
	deploy:
	  # Generate runtime environment variables
	  - uses: file/createOrUpdateEnvironmentFile
	    with:
	      target: ./.env
	    envs:
	      MicrosoftAppId: ${{AAD_APP_CLIENT_ID}}
	      MicrosoftAppPassword: ${{SECRET_AAD_APP_CLIENT_SECRET}}
	      STORAGE_ACCOUNT_CONNECTION_STRING: ${{SECRET_STORAGE_ACCOUNT_CONNECTION_STRING}}
	```

Reply:
Fixed